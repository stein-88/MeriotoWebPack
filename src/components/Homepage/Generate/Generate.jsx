import React from 'react'
import pc from '@images/homepc.png'
import schema from './data'
import s from './Generate.scss'

const Generate = () => {
	const { titulo, text } = schema
	return (
		<div className="row">
			<div className="col-12 text-muted">
				{titulo && <h2 className="h1 py-5 mb-0">{titulo}</h2>}
				<div className="col-12">
					<section className={`w-100 ${s.blocopc}`}>
						<div className="row">
							<div className="col-12 col-xl-8 col-lg-6 py-4 text-white">
								{text && text.length > 0 && text.map((ele, i) => (
									<p key={`generate ${i + 1}`} className="h3 mb-4">{ele}</p>
								))}
							</div>
							{pc
								&& (
									<div className="col-12 col-xl-4 col-lg-6">
										<img className="img-fluid w-100" src={pc} alt="pc" />
									</div>
								)}
						</div>
					</section>
				</div>
			</div>
		</div>
	)
}

export default Generate
