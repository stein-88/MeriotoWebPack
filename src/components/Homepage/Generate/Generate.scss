.blocopc {
  background-image: linear-gradient(90deg, #7e7e7e 80%, white 50%);
  padding: 0 20px;
  border-radius: 25px;
}
