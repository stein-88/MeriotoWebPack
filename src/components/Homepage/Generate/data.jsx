const schema = {
    titulo: 'Generate Impact',
    text: [
        'Our marketplace allows you to effortlessly manage and track your items, reduce costs, increase transparency and generate ESG data for your reporting.',
        'We offer external reuse via our marketplace, internal reuse and loaning asset inventory system.',
    ],
}

export default schema
