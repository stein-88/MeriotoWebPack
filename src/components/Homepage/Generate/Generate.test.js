import React from 'react'
import renderer from 'react-test-renderer'
import Generate from './Generate'
import schema from './data'
const { text } = schema

describe('testing Generate', () => {
    const compGenerate = renderer.create(<Generate />)
    test('Should render without props', () => {
        const jsonGenerate = compGenerate.toJSON()
        expect(jsonGenerate).toMatchSnapshot()
    })
    test('Should have 1 img, 1 title and 2 texts', () => {
        const resultTxt = compGenerate.root.findAllByProps({ className: 'h3 mb-4' })
        const resultTitle = compGenerate.root.findAllByProps({ className: 'h1 py-5 mb-0' })
        const resultimg = compGenerate.root.findAllByProps({ className: 'img-fluid w-100' })
        expect(resultTxt).toHaveLength(text.length)
        expect(resultTitle).toHaveLength(1)
        expect(resultimg).toHaveLength(1)
    })
})