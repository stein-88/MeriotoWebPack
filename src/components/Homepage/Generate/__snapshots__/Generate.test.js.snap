// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Generate Should render without props 1`] = `
<div
  className="row"
>
  <div
    className="col-12 text-muted"
  >
    <h2
      className="h1 py-5 mb-0"
    >
      Generate Impact
    </h2>
    <div
      className="col-12"
    >
      <section
        className="w-100 blocopc"
      >
        <div
          className="row"
        >
          <div
            className="col-12 col-xl-8 col-lg-6 py-4 text-white"
          >
            <p
              className="h3 mb-4"
            >
              Our marketplace allows you to effortlessly manage and track your items, reduce costs, increase transparency and generate ESG data for your reporting.
            </p>
            <p
              className="h3 mb-4"
            >
              We offer external reuse via our marketplace, internal reuse and loaning asset inventory system.
            </p>
          </div>
          <div
            className="col-12 col-xl-4 col-lg-6"
          >
            <img
              alt="pc"
              className="img-fluid w-100"
              src="test-file-stub"
            />
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
